src/python/pants/option/global_options.py

```diff
@@ -161,9 +161,15 @@ def register_bootstrap_options(cls, register):
              help='Timeout in seconds for URL reads when fetching binary tools from the '
                   'repos specified by --baseurls.')
     register('--binaries-path-by-id', type=dict, advanced=True,
-             help=('Maps output of uname for a machine to a binary search path. e.g. '
-                   '{("darwin", "15"): ["mac", "10.11"]), ("linux", "arm32"): ["linux"'
-                   ', "arm32"]}'))
+             help=("Maps output of uname for a machine to a binary search path: "
+                   "(sysname, id) -> (os, arch), e.g. {('darwin', '15'): ('mac', '10.11'), "
+                   "('linux', 'arm32'): ('linux', 'arm32')}."))
+    # BinaryTool options.
+    register('--force-baseurls', type=bool, default=False, advanced=True,
```

Comment:
I don't love this. It's weird to provide a really neat API, and then have this out-of-band way of overriding it, that requires an elaborate comment in the interface, and that will be completely non-obvious to the reader/debugger of this code.

I don't immediately have a better idea. But at the very least, let's name this something more obvious, like "--allow-external-binary-tool-downloads" (with a default of True, of course).

And then you can change the name of `url_generator()` to `get_external_url_generator()`, to clarify that it is affected by this option, and that if it returns None then "internal" (i.e., pants-hosted, or org-hosted) urls are used.

Reply:
Yeah, the only reason for this was for migration purposes, which isn't great. I would much prefer / tried adding an option to do this for individual `BinaryTool`s, but then you would just have to override every one of them (if running pants on an internal network). As a pants.ini option, I might not use it, but as a command-line option I can see it being useful. I agree on not liking it and there is probably a better way.

I'll implement the option name change and the method name change you described.

Reply:
c8bb85d has the changes. I think I got all the references to the old option, but we will see what CI has to say.